Modules/ZDC/src/ZDCRawDataTask.cxx

```diff
@@ -81,15 +81,49 @@ void ZDCRawDataTask::startOfCycle()
 
 void ZDCRawDataTask::monitorData(o2::framework::ProcessingContext& ctx)
 {
+  o2::framework::DPLRawParser parser(ctx.inputs());
+  int PayloadPerGBTW = 10;
+  int dataFormat;
+  const uint32_t* gbtw;
+  uint64_t count = 0;
+  size_t payloadSize;
+  size_t offset;
+  static uint64_t nErr[3] = { 0 };
 
-  o2::framework::InputRecord& inputs = ctx.inputs();
-  o2::framework::DPLRawParser parser(inputs);
   for (auto it = parser.begin(), end = parser.end(); it != end; ++it) {
-    auto const* payload = it.data();
-    size_t payloadSize = it.size();
-    if (payload != nullptr) {
-      for (int32_t ip = 0; ip < (int32_t)payloadSize; ip += 16) {
-        processWord((const uint32_t*)&payload[ip]);
+    auto rdhPtr = reinterpret_cast<const o2::header::RDHAny*>(it.raw());
+    if (rdhPtr == nullptr || !o2::raw::RDHUtils::checkRDH(rdhPtr, true)) {
+      nErr[0]++;
+      if (nErr[0] < 5) {
+        LOG(warning) << "ZDCDataReaderDPLSpec::run - Missing RAWDataHeader on page " << count;
```

Comment:
Having logging in `monitorData` is very risky. This method is called many times per seconds and it could start flooding the logs. Could you put it debug or make sure it is not called more than X per minute ?

Reply:
ok I removed the messages.